src/client.rs

```diff
@@ -47,21 +49,123 @@ pub struct Client {
+            ::hyper::client::Pool::with_connector(
+                Default::default(),
+                ::hyper::net::HttpsConnector::new(tls_client)));
+
+        hyper_client.set_redirect_policy(::hyper::client::RedirectPolicy::FollowNone);
+
+        Client {
             inner: Arc::new(ClientRef {
-                hyper: RwLock::new(client),
+                hyper: RwLock::new(hyper_client),
                 redirect_policy: Mutex::new(RedirectPolicy::default()),
                 auto_referer: AtomicBool::new(true),
                 auto_ungzip: AtomicBool::new(true),
             }),
+        }
+    }
+}
+
+/// A `ClientBuilder` can be used to a `Client` with a custom TLS configuration:
+///
+/// - with hostname verification disabled
+/// - with one or multiple custom certificates
+///
+/// # Examples
+///
+/// ```rust,no_run
+/// extern crate hyper_native_tls;
+/// extern crate reqwest;
+///
+/// use std::fs::File;
+/// use std::io::Read;
+///
+/// use hyper_native_tls::Certificate;
+/// use reqwest::ClientBuilder;
+///
+/// fn main() {
+///     // get a client builder
+///     let mut client_builder = ClientBuilder::new().unwrap();
+///
+///     // read a local binary DER encoded certificate
+///     let mut buf = Vec::new();
+///     let _ = File::open("my-cert.der").unwrap().read_to_end(&mut buf).unwrap();
+///
+///     // create a certificate
+///     let cert = Certificate::from_der(&buf).unwrap();
+///
+///     // add the certificate
+///     client_builder.add_root_certificate(cert).unwrap();
+///
+///     // create the actual client
+///     let client = client_builder.build().unwrap();
+///
+///     // [...]
+/// }
+/// ```
+///
+pub struct ClientBuilder {
+    inner: NativeTlsClientBuilder,
+    hostname_verification: bool,
+}
+
+impl ClientBuilder {
+    /// Constructs a new `ClientBuilder`
+    pub fn new() -> ::Result<ClientBuilder> {
+        let inner_tls_client = try_!(
+            NativeTlsClient::builder()
+                .map_err(|e| ::hyper::Error::Ssl(Box::new(e))));
+        Ok(ClientBuilder {
+            inner: inner_tls_client,
+            hostname_verification: true,
         })
     }
 
+    /// Returns a client
+    pub fn build(self) -> ::Result<Client> {
+        let mut tls_client = try_!(
+            self.inner.build().map_err(|e| ::hyper::Error::Ssl(Box::new(e))));
+        if ! self.hostname_verification {
+            tls_client.dannger_disable_hostname_verification(true);
+        }
+        Ok(From::from(tls_client))
+    }
+
+    /// Add a custom root certificate. This can be used to connect to a server that has a
+    /// self-signed certificate for example.
+    pub fn add_root_certificate(&mut self, cert: Certificate) -> ::Result<&mut ClientBuilder> {
+        try_!(self.inner.add_root_certificate(cert)
+                .map_err(|e| ::hyper::Error::Ssl(Box::new(e))));
+        Ok(self)
+    }
+
+    /// Disable hostname verification
+    pub fn disable_hostname_verification(&mut self) {
```

Comment:
Probably should have a scary name like the others, `danger_disable_hostname_verification`.